### PR TITLE
Add /apply option to write /fixall changes back to disk

### PR DESCRIPTION
**Pull request by sharwell**
This update to StyleCopTester allows it to be used to automate the correction of specific style violations in a solution.

Currently only one type of diagnostic can be fixed at a time, but it would be easy to create a script that invokes the tool multiple times to cover a set of diagnostics.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCopTester/Program.cs [hunk 1]
```diff
@@ -195,7 +212,24 @@ private static void WriteDiagnosticResults(ImmutableArray<Tuple<ProjectId, Diagn
                 {
                     stopwatch.Restart();
                     Console.WriteLine($"Calculating fix for {fix.CodeFixEquivalenceKey} using {fix.FixAllProvider} for {fix.NumberOfDiagnostics} instances.");
-                    await fix.GetOperationsAsync(cancellationToken).ConfigureAwait(true);
+                    var operations = await fix.GetOperationsAsync(cancellationToken).ConfigureAwait(true);
+                    if (applyChanges)
+                    {
+                        var applyOperations = operations.OfType<ApplyChangesOperation>().ToList();
+                        if (applyOperations.Count > 1)
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: @sharwell Do you know a case where this can happen?

**Comment by pdelvo** [comment]
:+1:

File: StyleCop.Analyzers/StyleCopTester/Program.cs (hunk 1 above)
Comment (sharwell) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: I do not. Presumably if this message ever appears someone will report it.

**Comment by oatkins** [comment]
Excellent! This will be really helpful. :+1:

**Pull request merged**